### PR TITLE
HPCC-31025 Add support for multiple trace exporters

### PR DESCRIPTION
**Pull request by ghalliday**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
@rpastrana I realised I didn't update the schema or readme to include "exporters".  I will do that later.

**Comment by ghalliday** [comment]
Another problem is that exporter: type=none should suppress the creation of the jlog exporter.

**Comment by ghalliday** [comment]
see notes on implementation in the jida

File: helm/hpcc/values.schema.json [hunk 1]
```diff
@@ -1123,16 +1123,10 @@
...
-          "type": "object",
-          "properties": {
-            "type": {
-              "type": "string",
-              "enum": ["batch", "simple"],
-              "description": "Defines the manner in which trace data is processed - in batches, or simple as available"
+            },
+            "batch": {
+              "type": "boolean",
+              "description": "If true, trace data is processed in a batch, if false when it is available"
```
Comment (rpastrana) [inline helm/hpcc/values.schema.json:1129]: Minor, but the last part doesn't make sense to me:  ", if false when it is available"

File: system/jlib/jtrace.cpp [hunk 2]
```diff
@@ -1052,161 +1054,163 @@ IProperties * getSpanContext(const ISpan * span)
...
+{
+    auto exporter = createExporter(exportConfig);
+    if (!exporter)
+        return nullptr;
+
+    if (exportConfig->getPropBool("batch/@enabled", false))
+    {
+        //Groups several spans together, before sending them to an exporter.
+        //MORE: These options should be configurable from batch/@option
+        opentelemetry::v1::sdk::trace::BatchSpanProcessorOptions options; //size_t max_queue_size = 2048;
```
Comment (rpastrana) [inline system/jlib/jtrace.cpp:1174]: https://track.hpccsystems.com/browse/HPCC-31040

**Review by rpastrana — COMMENTED** [review]
@ghalliday looks good.
I'm not crazy about the config option name "createDefaultLogExporter" from the config point of view it should be more along the lines of "EnableDefaultLogExporter". subtle difference but I felt it was worth mentioning.
Also, there's a few sample config values files in helm/examples/tracing that would need to be updated, perhaps replaced:
- [jlog-collector-fulloutput.yaml](https://github.com/hpcc-systems/HPCC-Platform/blob/master/helm/examples/tracing/jlog-collector-fulloutput.yaml)
- [otlp-grcp-collector-default.yaml](https://github.com/hpcc-systems/HPCC-Platform/blob/master/helm/examples/tracing/otlp-grcp-collector-default.yaml)
- [otlp-http-collector-default.yaml](https://github.com/hpcc-systems/HPCC-Platform/blob/master/helm/examples/tracing/otlp-http-collector-default.yaml)
Adding a new one showing off multiple exporters might also be helpful

**Comment by ghalliday** [comment]
@rpastrana I think I have addressed your comments, and resolved the clashes with your GRPC PR which I merged.

**Review by rpastrana — APPROVED** [review]
@ghalliday looks good.

**Pull request merged**